get_info.php

```diff
@@ -201,56 +234,121 @@ function get_list_of_surveys(
-            $survey_data["questions"] = [];
+    } elseif ($survey_id) {
+        //invalid survey_id check
+        $valid_surveys = filter_surveys(
+            $survey_choices,
+            $bind_variables[':survey_id']
+        );
 
-            //no number locked by course when viewing admin survey
-            if (!isset($survey_data["number_locked_by_course"])) {
-                $survey_data["number_locked_by_course"] = null;
-            }
+        //get all surveys selected
+        $survey_data = get_query_result(gen_query_get_survey_data(), [
+            ":survey_id" => $valid_surveys
+        ]);
+        $precidence = array(
+            'is_admin' => [0],
+            'is_instructor' => [1, 0],
+            'is_ta' => [2, 1, 0]
+        );
+
+        //foreach survey_data combine the choices to create one choices attribute
+        foreach ($survey_data as $index => $value) {
+            $all_choices = [];
 
-            $responses = [];
+            array_push($all_choices, $value["dept_survey_choice_id"]);
+            array_push($all_choices, $value["course_survey_choice_id"]);
+            array_push($all_choices, $value["ta_survey_choice_id"]);
+            //set role
+            $ro = '';
+
+            //Used to select question on account
+            $count = 0;
+            // this is used to keep track of null columns and determine the override from the level above
+            $num_to_set = 6;
+
+            foreach ($role as $key => $value) {
+                if ($value == "TRUE") {
+                    $ro = $key;
+                    break;
+                }
+                $count += 2;
+                $num_to_set -= 2;
+            }
+            $survey_data[$index]["questions"] = [];
+            //get survey responses if the survey results are requested
             if ($is_instance) {
                 //get list of responses when survey_results is requested
                 $responses = get_query_result(gen_query_survey_responses(), [
                     ":survey_id" => $bind_variables[":survey_id"]
                 ]);
             }
 
-            // set response and position for each selected question
-            for ($j = 1; $j <= 6; $j++) {
-                $question = $list_of_quesitons[$survey_data["choice" . $j] - 1];
-                $question["responses"] = isset(
-                    $responses[$survey_data["choice" . $j] - 1]
-                )
-                    ? $responses[$survey_data["choice" . $j] - 1]["answers"]
-                    : null;
-                $question['position'] = $j;
-                array_push($survey_data["questions"], $question);
-                unset($survey_data["choice" . $j]);
+            //this part we want to join the choices together, so take the choices
+            //so use precidence to determine the column that we are looking at to build
+            //the survey
+            foreach ($precidence[$ro] as $key => $value) {
```

Comment:
This function is confusing. Just get the three choices objects and then blend them together. Don't check to see if the "locked" questions are correct. That should be done when we set the choices.